### PR TITLE
refactor(snap): Update command and metadata sourcing

### PR DESCRIPTION
**Pull request by MonicaisHer**
- Inline CLIs flags
- Remove unused packages
- Remove external metadata sourcing

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -2,7 +2,9 @@ name: edgex-device-rest
 base: core20
 type: app
 license: Apache-2.0
-adopt-info: metadata
+summary: EdgeX Device REST Service
```
Comment (farshidtz) [inline snap/snapcraft.yaml:5]: This isn't the codename but the summary. A better English summary is:
```suggestion
summary: EdgeX REST Device Service
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -60,7 +58,6 @@ parts:
       install -DT ./helper-go $SNAPCRAFT_PART_INSTALL/bin/helper-go
 
   device-rest:
-    after: [metadata]
     source: .
     plugin: make
     build-packages: [git, zip, pkg-config]
```
Comment (farshidtz) [inline snap/snapcraft.yaml:63]: These were probably needed for libzmq. No longer needed.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (MonicaisHer) [reply snap/snapcraft.yaml:63]: It appears that the git package is required, as an error is present without it:
```
/bin/bash: line 35: git: command not found
```

Comment (farshidtz) [reply snap/snapcraft.yaml]: Strange. I know that Go compiler needs it. Maybe it isn't included by the Go plugin on the core20 base.

**Review by farshidtz — APPROVED** [review]

**Pull request merged**